engine/schema/src/main/resources/META-INF/db/schema-41400to41500.sql

```diff
@@ -837,3 +837,6 @@ UPDATE `cloud`.`guest_os` SET category_id = 7 WHERE id = 275 AND display_name =
 -- Add flag 'hidden' in tables usage_ip_address and cloud_usage
 ALTER TABLE `cloud_usage`.`usage_ip_address` ADD COLUMN `is_hidden` smallint(1) NOT NULL DEFAULT '0' COMMENT 'is usage hidden';
 ALTER TABLE `cloud_usage`.`cloud_usage` ADD COLUMN `is_hidden` smallint(1) NOT NULL DEFAULT '0' COMMENT 'is usage hidden';
+
+-- Fix Zones are returned in a random order (#3934)
+UPDATE `cloud`.`data_center` JOIN (SELECT COUNT(1) AS count FROM `cloud`.`data_center` WHERE `sort_key` != 0) AS tbl_tmp SET `sort_key` = `id` WHERE count = 0;
```

Comment:
excuse my ignorance. why not just 
```
UPDATE `cloud`.`data_center` SET `sort_key` = `id` WHERE `sort_key` = 0;
```
this seems like it going to work as well, but i'm missing the point, i guess. at worst the first in the sortorder has to be put back at the fron, but this should be a onetime effort.

Reply:
@DaanHoogland assume the sort_key is 1,0, with your sql sort_key will be changed to 1,2.
the order of zones is changed by the sql.

Reply:
so in the case above, `sort_key` is only changed if there is no `sort_key != 0`. This would mean that if there are three where it is 0 and one where it is 10, nothing gets changed, does it?

if this is at all possible btw!

Reply:
@DaanHoogland exactly.
IMHO, if there is a non-zero sort_key , which mean cloudstack user might use sort_key for ordering, we will not touch it.
if all sort_key are 0, we change sort_key to id.

Reply:
ok, sounds reasonable.